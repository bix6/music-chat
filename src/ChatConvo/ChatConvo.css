.chat-convo {
    text-align: left;
    margin: 2rem auto;
    max-width: 600px;
    border: 1px solid black;
    overflow-y: auto;
    padding: 1rem;
}

.chat-convo-li {
    line-height: 2rem;
}

.chat-convo-li:nth-child(odd) {
    background-color: lightblue;
}