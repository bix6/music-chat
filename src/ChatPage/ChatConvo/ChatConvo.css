.chat-convo-div {
  text-align: left;
  overflow-y: auto;
}

.chat-convo-li {
  line-height: 2rem;
}

.chat-convo-li:nth-child(odd) {
  background-color: rgba(255, 255, 255, 0.6);
}

.chat-convo-li:nth-child(even) {
  background-color: rgba(255, 255, 255, 0.2);
}
