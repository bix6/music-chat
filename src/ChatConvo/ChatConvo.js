import React from 'react';
import './ChatConvo.css';
import ChatContext from '../ChatContext';

class ChatConvo extends React.Component {
    static contextType = ChatContext;

   createListItems = () => {
       const currentChatroom = this.context.currentChatroom;
       const messages = this.context.messages[currentChatroom];
       const listItems = messages.map((message, i) => {
           return <li className="chat-convo-li" key={ i }>{ message }</li>
       })
       return listItems;
   }

    render() {
        return (
            <div className='chat-convo'>
                <ol className='chat-convo-ol'>
                    { this.createListItems() }
                </ol>
            </div>
        )
    }
}

export default ChatConvo;