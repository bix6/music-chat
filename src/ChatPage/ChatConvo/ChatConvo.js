import React from "react";
import "./ChatConvo.css";

class ChatConvo extends React.Component {
  static defaultProps = {
    messages: {
      Loading: ["Bixbot: Chat loading... "],
    },
    currentChatroom: "Loading",
  };

  createMessageLi = (messageObj, i) => {
    const message = messageObj.username + ": " + messageObj.message;

    return (
      <li key={i} className="chat-convo-li">
        {message}
      </li>
    );
  };

  createYoutubeVideoLi = (messageObj, i) => {
    const src = `https://www.youtube.com/embed/${messageObj.contentId}`;
    const title = `Embeded YouTube videoId: ${messageObj.contentId}`;

    return (
      <li key={i} className="chat-convo-li">
        {messageObj.username + ": "}
        <iframe
          width="480px" // Fallback size; gets set dynamically in css
          height="270px"
          src={src}
          frameBorder="0"
          encrypted-media="true"
          gyroscope="true"
          picture-in-picture="true"
          title={title}
          allowFullScreen
        />
      </li>
    );
  };

  /* Create li elems for each messageObj in the current chatroom */
  createConvoListElems = () => {
    return this.props.messages[this.props.currentChatroom].map(
      (messageObj, i) => {
        if (messageObj.contentType === "text") {
          return this.createMessageLi(messageObj, i);
        } else if (messageObj.contentType === "youtube video") {
          return this.createYoutubeVideoLi(messageObj, i);
        }
        return "Bixbot: Error Retrieving Message";
      }
    );
  };

  render() {
    return (
      <div className="chat-convo-div">
        <ol>{this.createConvoListElems()}</ol>
      </div>
    );
  }
}

export default ChatConvo;
