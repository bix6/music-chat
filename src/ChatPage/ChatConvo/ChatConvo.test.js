import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';
import ChatConvo from './ChatConvo';

describe('ChatConvo Component', () => {
    it('Smoke Test: Renders Default', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <ChatConvo />
            </BrowserRouter>,
            div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('Snapshot Test: Default', () => {
        const wrapper = shallow(<ChatConvo />);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})