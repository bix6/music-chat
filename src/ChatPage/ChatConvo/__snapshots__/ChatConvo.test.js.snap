// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatConvo Component Snapshot Test: Default 1`] = `
<div
  className="chat-convo-div"
>
  <ol />
</div>
`;
